src/structures/GuildScheduledEvent.js

```diff
@@ -0,0 +1,416 @@
+    return this.client.channels.resolve(this.channelId);
+  }
+
+  /**
+   * The guild this scheduled event belongs to
+   * @type {?Guild}
+   * @readonly
+   */
+  get guild() {
+    return this.client.guilds.resolve(this.guildId);
+  }
+
+  /**
+   * The URL to the guild scheduled event
+   * @type {string}
+   * @readonly
+   */
+  get url() {
+    return Endpoints.scheduledEvent(this.client.options.http.scheduledEvent, this.guildId, this.id);
+  }
+
+  /**
+   * Options used to create an invite URL to a {@link GuildScheduledEvent}
+   * @typedef {CreateInviteOptions} CreateGuildScheduledEventInviteURLOptions
+   * @property {GuildInvitableChannelResolvable} [channel] The channel to create the invite in.
+   * <warn>This is required when the `entityType` of `GuildScheduledEvent` is `EXTERNAL`, gets ignored otherwise</warn>
+   */
+
+  /**
+   * Creates an invite URL to this guild scheduled event.
+   * @param {CreateGuildScheduledEventInviteURLOptions} [options] The options to create the invite
+   * @returns {Promise<string>}
+   */
+  async createInviteURL(options) {
+    let channelId = this.channelId;
+    if (this.entityType === 'EXTERNAL') {
+      if (!options?.channel) throw new Error('INVITE_OPTIONS_MISSING_CHANNEL');
+      channelId = this.guild.channels.resolveId(options.channel);
+      if (!channelId) throw new Error('GUILD_CHANNEL_RESOLVE');
+    }
+    const invite = await this.guild.invites.create(channelId, options);
+    return Endpoints.invite(this.client.options.http.invite, invite.code, this.id);
+  }
+
+  /**
+   * Edits this guild scheduled event.
+   * @param {GuildScheduledEventEditOptions} options The options to edit the guild scheduled event
+   * @returns {Promise<GuildScheduledEvent>}
+   * @example
+   * // Edit a guild scheduled event
+   * guildScheduledEvent.edit({ name: 'Party' })
+   *  .then(guildScheduledEvent => console.log(guildScheduledEvent))
+   *  .catch(console.error);
+   */
+  edit(options) {
+    return this.guild.scheduledEvents.edit(this.id, options);
+  }
+
+  /**
+   * Deletes this guild scheduled event.
+   * @returns {Promise<GuildScheduledEvent>}
+   * @example
+   * // Delete a guild scheduled event
+   * guildScheduledEvent.delete()
+   *  .then(guildScheduledEvent => console.log(guildScheduledEvent))
+   *  .catch(console.error);
+   */
+  async delete() {
+    await this.guild.scheduledEvents.delete(this.id);
+    const clone = this._clone();
+    return clone;
+  }
+
+  /**
+   * Sets a new name for the guild scheduled event.
+   * @param {string} name The new name of the guild scheduled event
+   * @returns {Promise<GuildScheduledEvent>}
+   * @example
+   * // Set name of a guild scheduled event
+   * guildScheduledEvent.setName('Birthday Party')
+   *  .then(guildScheduledEvent => console.log(`Set the name to: ${guildScheduledEvent.name}`))
+   *  .catch(console.error);
+   */
+  setName(name) {
+    return this.edit({ name, entityMetadata: { location: this.location } });
```

Comment:
this.location doesn't exist. Did you mean `entityMetadata: this.entityMetadata`?

Reply:
this was a dirtttty hack that I forgot to update, fortunately, it's not required anymore. So, I'll just remove it.